ray-operator/controllers/ray/raycluster_controller_test.go

```diff
@@ -190,9 +189,39 @@ var _ = Context("Inside the default namespace", func() {
 				time.Second*15, time.Millisecond*500).Should(BeNil(), "autoscaler RoleBinding = %v", rbName)
 		})
 
+		It("should be able to update all Pods to Running", func() {
+			for _, workerPod := range workerPods.Items {
+				workerPod.Status.Phase = corev1.PodRunning
+				Expect(k8sClient.Status().Update(ctx, &workerPod)).Should(BeNil())
+			}
+			Consistently(
+				listResourceFunc(ctx, &workerPods, workerFilterLabels, &client.ListOptions{Namespace: "default"}),
+				time.Second*5, time.Millisecond*500).Should(Equal(3), fmt.Sprintf("workerGroup %v", workerPods.Items))
+			for _, workerPod := range workerPods.Items {
+				Expect(workerPod.Status.Phase).Should(Equal(corev1.PodRunning))
+			}
+
+			for _, headPod := range headPods.Items {
+				headPod.Status.Phase = corev1.PodRunning
+				Expect(k8sClient.Status().Update(ctx, &headPod)).Should(BeNil())
+			}
+			Consistently(
+				listResourceFunc(ctx, &headPods, headFilterLabels, &client.ListOptions{Namespace: "default"}),
+				time.Second*5, time.Millisecond*500).Should(Equal(1), fmt.Sprintf("headGroup %v", headPods.Items))
+			for _, headPod := range headPods.Items {
+				Expect(headPod.Status.Phase).Should(Equal(corev1.PodRunning))
+			}
+		})
```

Comment:
Need to manually update Pod statuses otherwise they'll always be Pending. [envtest](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest) doesn't create a full K8s cluster. It's only the control plane. There's no container runtime or any other K8s controllers. So Pods are created, but no controller updates them from Pending to Running.

> setting up and starting an instance of etcd and the Kubernetes API server, without kubelet, controller-manager or other components 

https://book.kubebuilder.io/reference/envtest.html

Reply:
Cool!